File: packages/ios/Sources/Components/Icon/Icon.swift
import SwiftUI

public enum KozmosIconSize {
    case xs
    case sm
    case md
    case lg
    case xl

    var pointSize: CGFloat {
        switch self {
        case .xs: return 12
        case .sm: return 16
        case .md: return 20
        case .lg: return 24
        case .xl: return 32
        }
    }
}

public enum KozmosIconColor {
    case `default`
    case muted
    case primary
    case destructive

    var swiftUIColor: Color {
        switch self {
        case .default: return KozmosColors.primitivesColorsForeground100
        case .muted: return KozmosColors.primitivesColorsForeground500
        case .primary: return KozmosColors.primitivesColorsTheme500
        case .destructive: return KozmosColors.primitivesColorsEmotionalDanger600
        }
    }
}

public struct KozmosIcon: View {
    let name: String
    let size: KozmosIconSize
    let color: KozmosIconColor
    let accessibilityLabel: String?

    public init(
        _ name: String = "home-line",
        size: KozmosIconSize = .md,
        color: KozmosIconColor = .default,
        accessibilityLabel: String? = nil
    ) {
        self.name = name
        self.size = size
        self.color = color
        self.accessibilityLabel = accessibilityLabel
    }

    public var body: some View {
        Image(systemName: Self.symbolName(for: name))
            .font(.system(size: size.pointSize, weight: .medium))
            .foregroundColor(color.swiftUIColor)
            .frame(width: size.pointSize, height: size.pointSize)
            .accessibilityHidden(accessibilityLabel == nil)
            .accessibilityLabel(accessibilityLabel ?? "")
    }

    static func symbolName(for rawName: String) -> String {
        let name = resolvedIconName(rawName)

        switch name {
        case "activity": return "waveform.path.ecg"
        case "alert-circle": return "exclamationmark.circle"
        case "alert-triangle": return "exclamationmark.triangle"
        case "arrow-left": return "arrow.left"
        case "arrow-right": return "arrow.right"
        case "bell-01": return "bell"
        case "building-01": return "building.2"
        case "bus": return "bus"
        case "calendar": return "calendar"
        case "check": return "checkmark"
        case "chevron-down": return "chevron.down"
        case "chevron-left": return "chevron.left"
        case "chevron-right": return "chevron.right"
        case "chevron-up": return "chevron.up"
        case "clock": return "clock"
        case "compass-01": return "compass"
        case "download-01": return "arrow.down.to.line"
        case "edit-01": return "pencil"
        case "home-line": return "house"
        case "info-circle": return "info.circle"
        case "lock-01": return "lock"
        case "map-01": return "map"
        case "marker-pin-01": return "mappin"
        case "menu-01": return "line.3.horizontal"
        case "minus": return "minus"
        case "navigation-pointer-01": return "location.north"
        case "plus": return "plus"
        case "qr-code-01": return "qrcode"
        case "route": return "point.topleft.down.curvedto.point.bottomright.up"
        case "scan": return "viewfinder"
        case "search-md": return "magnifyingglass"
        case "settings-01": return "gearshape"
        case "trash-01": return "trash"
        case "upload-01": return "arrow.up.to.line"
        case "user-01": return "person"
        case "users-01": return "person.2"
        case "wifi": return "wifi"
        case "x-close": return "xmark"
        default: return "questionmark.circle"
        }
    }

    private static func resolvedIconName(_ name: String) -> String {
        switch name {
        case "add": return "plus"
        case "back": return "arrow-left"
        case "close", "x": return "x-close"
        case "delete": return "trash-01"
        case "edit": return "edit-01"
        case "home": return "home-line"
        case "location", "map-pin": return "marker-pin-01"
        case "menu": return "menu-01"
        case "next": return "arrow-right"
        case "notifications": return "bell-01"
        case "search": return "search-md"
        case "settings": return "settings-01"
        case "user": return "user-01"
        case "users": return "users-01"
        case "warning": return "alert-triangle"
        default: return name
        }
    }
}

public typealias Icon = KozmosIcon

File: packages/android/src/main/java/com/kozmos/components/Icon/Icon.kt
package com.kozmos.components.icon

import androidx.compose.foundation.layout.size
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.ArrowBack
import androidx.compose.material.icons.automirrored.filled.ArrowForward
import androidx.compose.material.icons.automirrored.filled.KeyboardArrowLeft
import androidx.compose.material.icons.automirrored.filled.KeyboardArrowRight
import androidx.compose.material.icons.automirrored.filled.ShowChart
import androidx.compose.material.icons.filled.AccessTime
import androidx.compose.material.icons.filled.Add
import androidx.compose.material.icons.filled.Business
import androidx.compose.material.icons.filled.CalendarToday
import androidx.compose.material.icons.filled.Check
import androidx.compose.material.icons.filled.Close
import androidx.compose.material.icons.filled.Delete
import androidx.compose.material.icons.filled.DirectionsBus
import androidx.compose.material.icons.filled.Download
import androidx.compose.material.icons.filled.Edit
import androidx.compose.material.icons.filled.Explore
import androidx.compose.material.icons.filled.FileUpload
import androidx.compose.material.icons.filled.Home
import androidx.compose.material.icons.filled.Info
import androidx.compose.material.icons.filled.KeyboardArrowDown
import androidx.compose.material.icons.filled.KeyboardArrowUp
import androidx.compose.material.icons.filled.Lock
import androidx.compose.material.icons.filled.Map
import androidx.compose.material.icons.filled.Menu
import androidx.compose.material.icons.filled.Navigation
import androidx.compose.material.icons.filled.Notifications
import androidx.compose.material.icons.filled.Person
import androidx.compose.material.icons.filled.PersonAddAlt
import androidx.compose.material.icons.filled.Place
import androidx.compose.material.icons.filled.QrCode2
import androidx.compose.material.icons.filled.Remove
import androidx.compose.material.icons.filled.Route
import androidx.compose.material.icons.filled.Search
import androidx.compose.material.icons.filled.Settings
import androidx.compose.material.icons.filled.Upload
import androidx.compose.material.icons.filled.Warning
import androidx.compose.material.icons.filled.Wifi
import androidx.compose.material3.Icon
import androidx.compose.material3.LocalContentColor
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import com.kozmos.tokens.KozmosColors

enum class KozmosIconSize(val dp: Dp) {
    Xs(12.dp),
    Sm(16.dp),
    Md(20.dp),
    Lg(24.dp),
    Xl(32.dp)
}

enum class KozmosIconColor {
    Default,
    Muted,
    Primary,
    Destructive
}

@Composable
fun KozmosIcon(
    name: String = "home-line",
    modifier: Modifier = Modifier,
    size: KozmosIconSize = KozmosIconSize.Md,
    color: KozmosIconColor = KozmosIconColor.Default,
    contentDescription: String? = null
) {
    Icon(
        imageVector = resolveIconVector(name),
        contentDescription = contentDescription,
        tint = resolveIconTint(color),
        modifier = modifier.size(size.dp)
    )
}

private fun resolveIconName(name: String): String = when (name) {
    "add" -> "plus"
    "back" -> "arrow-left"
    "close", "x" -> "x-close"
    "delete" -> "trash-01"
    "edit" -> "edit-01"
    "home" -> "home-line"
    "location", "map-pin" -> "marker-pin-01"
    "menu" -> "menu-01"
    "next" -> "arrow-right"
    "notifications" -> "bell-01"
    "search" -> "search-md"
    "settings" -> "settings-01"
    "user" -> "user-01"
    "users" -> "users-01"
    "warning" -> "alert-triangle"
    else -> name
}

private fun resolveIconVector(rawName: String): ImageVector {
    return when (resolveIconName(rawName)) {
        "activity" -> Icons.AutoMirrored.Filled.ShowChart
        "alert-circle" -> Icons.Default.Info
        "alert-triangle" -> Icons.Default.Warning
        "arrow-left" -> Icons.AutoMirrored.Filled.ArrowBack
        "arrow-right" -> Icons.AutoMirrored.Filled.ArrowForward
        "bell-01" -> Icons.Default.Notifications
        "building-01" -> Icons.Default.Business
        "bus" -> Icons.Default.DirectionsBus
        "calendar" -> Icons.Default.CalendarToday
        "check" -> Icons.Default.Check
        "chevron-down" -> Icons.Default.KeyboardArrowDown
        "chevron-left" -> Icons.AutoMirrored.Filled.KeyboardArrowLeft
        "chevron-right" -> Icons.AutoMirrored.Filled.KeyboardArrowRight
        "chevron-up" -> Icons.Default.KeyboardArrowUp
        "clock" -> Icons.Default.AccessTime
        "compass-01" -> Icons.Default.Explore
        "download-01" -> Icons.Default.Download
        "edit-01" -> Icons.Default.Edit
        "home-line" -> Icons.Default.Home
        "info-circle" -> Icons.Default.Info
        "lock-01" -> Icons.Default.Lock
        "map-01" -> Icons.Default.Map
        "marker-pin-01" -> Icons.Default.Place
        "menu-01" -> Icons.Default.Menu
        "minus" -> Icons.Default.Remove
        "navigation-pointer-01" -> Icons.Default.Navigation
        "plus" -> Icons.Default.Add
        "qr-code-01" -> Icons.Default.QrCode2
        "route" -> Icons.Default.Route
        "scan" -> Icons.Default.Search
        "search-md" -> Icons.Default.Search
        "settings-01" -> Icons.Default.Settings
        "trash-01" -> Icons.Default.Delete
        "upload-01" -> Icons.Default.Upload
        "user-01" -> Icons.Default.Person
        "users-01" -> Icons.Default.PersonAddAlt
        "wifi" -> Icons.Default.Wifi
        "x-close" -> Icons.Default.Close
        else -> Icons.Default.Info
    }
}

@Composable
private fun resolveIconTint(color: KozmosIconColor): Color = when (color) {
    KozmosIconColor.Default -> LocalContentColor.current
    KozmosIconColor.Muted -> KozmosColors.primitivesColorsForeground500
    KozmosIconColor.Primary -> KozmosColors.primitivesColorsTheme500
    KozmosIconColor.Destructive -> KozmosColors.primitivesColorsEmotionalDanger600
}
